cmake/UnixCompileRules.cmake

```
@@ -0,0 +1,3 @@
+set (CMAKE_CXX_FLAGS_COMMON "-Wall -Wextra")
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_COMMON} -O1 -g")
```

The `-Og` level specifies we want to run optimizations that don't degrade debuggability (similar to -O1)
```suggestion
set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_COMMON} -Og -g")
```